### PR TITLE
[Security] fixed DebugAccessDecisionManager::setVoters()

### PR DESCRIPTION
**Pull request by HeahDude**
| Q | A |
| --- | --- |
| Branch? | 3.1 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | https://github.com/schmittjoh/JMSSecurityExtraBundle/issues/207 |
| License | MIT |
| Doc PR | ~ |

### CODE REVIEW

**Comment by HeahDude** [comment]
Ref https://github.com/schmittjoh/JMSSecurityExtraBundle/issues/207#issuecomment-230101364

**Comment by WouterJ** [comment]
👍 This check makes much more sense imo

Status: reviewed

**Comment by HeahDude** [comment]
Tests are failing :(

Status: Needs Work

File: src/Symfony/Component/Security/Core/Authorization/DebugAccessDecisionManager.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ public function decide(TokenInterface $token, array $attributes, $object = null)
      */
     public function setVoters(array $voters)
     {
-        if (!$this->manager instanceof AccessDecisionManager) {
+        if (!method_exists($this->voters, 'setVoters')) {
```
Comment (chalasr) [inline src/Symfony/Component/Security/Core/Authorization/DebugAccessDecisionManager.php]: Am I missing something or you are calling `method_exists` on an array?
Should it not be `!method_exists($this->manager, 'setVoters')` instead?

Comment (HeahDude) [inline src/Symfony/Component/Security/Core/Authorization/DebugAccessDecisionManager.php]: Of courses! Went to quickly on this one, thanks!

**Comment by HeahDude** [comment]
Status: Needs Review

**Comment by fabpot** [comment]
Thank you @HeahDude.

**Pull request merged**